Changes performed on English documents to help po4a

smoe opened the pull request:
These are mostly addressing the indentation. A few artifacts
from franctic bug searches are possible but should be an exception.

A few blobs sneaked in that have not yet been translated from French or Spanish to English, yet, but that felt like these should just be missing when the automated generation of _fr and _es files are implemented.The copying was also required to match the structure for the .po generation. I suggest to discuss the value and proper translations of these smallish fragments to English in follow-up commits. I had preferred to leave these contributions in the original language to have them easily detectable than to come up with something that may resemble English at a first glance.

By @siropolis there will be more changes coming from the files that have diverged more from the original. The .po files will be derived from versions that have these changes already applied.

smoe commented:
I ran into difficulties since when paragraphs were copied between languages then the reference was often also translated and then did not exist in the destination. Took too long to fix. Wrt weblate it is annoying how many badly formatted itemizations there still are. Weblate will then present the line with the "*" separate from the following lines. I hence stronlgy suggest to fix everything that seems weird in weblate in the English source tree first.

smoe commented:
@SebKuzminsky  , @silopolis and @petterreinholdtsen  ,  for this PR I went through a series of files - too many it feels - with my experience from what po4a can understand as a single block from when I synced the _es and _fr files. The changes I had made to the English files to get any sync between the languages are also included.  This is admittedly not perfect, still, but I have some good confidence that it is better than what we had before. May I say much better?

The .adoc to .html transition is very sensitive when it comes to references. Quite some bits are not optimal in the routine the documents have that shall be fixed when the .adoc guidelines have evolved and are agreed on. For instance, the [[anchors]] need to stand directly in front of the section headers (after the "=" characters). And the (((index))) entries should be directly in front of the content at the beginning of the paragraph. I suggest to postpone this effort, though and just get something out that po4a can (likely) work with.

The (w)diff in github is much easier to read than via the command line. Since quite a few commits are changing the same parts of the text, I am preparing a squashed rebased and will push again. There is nothing functional, just text diffs, to I think this is to be preferred.

smoe commented:
I had removed another "all-English" Spanish document which had the consequence that some links from other Spanish files were stale. Also, against what I summarized from https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#anchor, the [[...]]s on in the section headers fail to give compatible references. So I was reverting these few "test cases".

smoe commented:
@SebKuzminsky  - I have admittedly few ideas how you want to review all this without investing a good fraction of the time that I have invested to prepare this all.  I suggest you look at 10 or 20 files so you know what I was doing and have some confidence that I have not messed things up completely. The experiences we make with weblate will be most instructional. You will find that I have introduced a few "pilots" if you like. For instance for a few files/paragraphs I have rearranged paragraphs to have one sentence per line. This may be of considered beneficial for the translation, especially since weblate expects the same number of lines to remain invariant.

Besides some work on consistent ordering of [[anchors]] and (((indices))), the changes should be mostly minimal and only white-space, though. Little text if any will be changed. An exception seems to be some description on the translations which I admit to have forgotten about. Anyway.

@silopolis I hope for some feedback from @SebKuzminsky on how much he likes this or not and then we know if your changes should go into master prior to mine.

smoe commented:
@SebKuzminsky I just rebased to master to integrate all those latest changes to the documentation. Please kindly give me a path forward on what needs to be done to get this reviewed or even merged.

silopolis commented:
Le mer. 9 mars 2022 à 01:12, Steffen Möller ***@***.***> a
écrit :

> @SebKuzminsky <https://github.com/SebKuzminsky> I just rebased to master
> to integrate all those latest changes to the documentation. Please kindly
> give me a path forward on what needs to be done to get this reviewed or
> even merged?
>

That's the one I should review, right?

smoe commented:
@silopolis Ideally, we would have the asciidoc styleguides and thus have something resembling an objective measure to decide what the asciidoc file should look like. What is happening now is that I basically tell every new commit that this commit is right since I do not feel like checking what the exact change is. A typo fixed in a table that I had rearranged will thus kill all my changes to the table. While I am not free from frustration over this situation, well, I need to blame myself to allow the patch to become that large. But then again, it is mostly just white space and local permutations of words, so nothing challenging to review, really.

silopolis commented:
Le mer. 9 mars 2022 à 11:22, Steffen Möller ***@***.***> a
écrit :

> @silopolis <https://github.com/silopolis> Ideally, we would have the
> asciidoc styleguides and thus have something resembling an objective
> measure to decide what the asciidoc file should look like.
>
I have started working on it and will spend some time on it again from now
on...

While I am not free from frustration over this situation, well, I need to
> blame myself to allow the patch to become that large.
>
No blame allowed, you did it and I don't see a problem as long as all mods
fix the same issue.

But then again, it is mostly just white space and local permutations of
> words, so nothing challenging to review, really.
>
OK will have a look may be start review at lunch and hopefully finish this
evening...

smoe commented:
Wow! This is great! Many thanks!
Concerning "only one issue per file" this is not feasible. The initial task was to get po4a to sync between the languages. So when the translation has done something better than you have backported that to English, even though the English was technically without error. Or you have removed blanks at the end of lines which may hit just anywhere. The history was folder/file-based. And at some point I had squashed diffs to reduce the work with conflicts when rebasing after updates.

What should not happen is that you invest all your time with the review and @SebKuzminsky not having expected that/being waiting for it.

Something that @petterreinholdtsen does not yet see like I see it is that I have fixed the (kind of) problems in this branch that he now sees in where po4a is stuck. So, for me, getting this in and the .po files updated accordingly is a precondition for me to actively involve myself in the pulling of the long po4a convoy out of the mud. It is hence not only myself crying after some invested weekends to bear some fruit but also to see the po4a going somewhere why I keep nagging about to this PR to get its chances.

silopolis commented:
Le mer. 9 mars 2022 à 11:56, Steffen Möller ***@***.***> a
écrit :

> Wow! This is great! Many thanks!
>
My pleasure mate!


Concerning "only one issue per file" this is not feasible.
>
I just meant that your PR may be huge, as long as it stays focused on
solving a coherent set of issues, I don't really see the problem. After
reaching half of it in review, I'd find dirty to have it split in any way.


What should not happen is that you invest all your time with the review and
> @SebKuzminsky <https://github.com/SebKuzminsky> not having expected
> that/being waiting for it.
>
That'd surely be a huge disappointment! We're are doing here a tedious
sanitizing job without which we should just forget the idea of bringing
documentation to another level...


Something that @petterreinholdtsen <https://github.com/petterreinholdtsen>
> does not yet see like I see it is that I have fixed the (kind of) problems
> in this branch that he now sees in where po4a is stuck. So, for me, getting
> this in and the .po files updated accordingly is a precondition for me to
> actively involve myself in the pulling of the long po4a convoy out of the
> mud. It is hence not only myself crying after some invested weekends to
> bear some fruit but also to see the po4a going somewhere why I keep nagging
> about to this PR to get its chances.
>
While reviewing your PR, I see me again unable to refrain making a
significant number of additional mods suggestions to try to get the src
english doc to a clean sound formating and asciidoc tagging.
Like I told you in matrix/irc PM, could you (and others) please make me a
halfway feedback to tell me what you think...

My next task will definitely be working on the docs writing guidelines to
submit them, discuss them, and have them approved and merged ASAP.

I won't invest any more time on this documentation code base before making
it a solid foundation for the future LinuxCNC documentation system (yes,
I'm looking at you Antora!)

Always glad to be part of this and to collaborate with you guys

Whishing you a beautiful we

TY
Jérémie

silopolis commented:
Le sam. 12 mars 2022 à 00:58, Steffen Möller ***@***.***> a
écrit :

> While reviewing your PR, I see me again unable to refrain making a
> significant number of additional mods suggestions to try to get the src
> english doc to a clean sound formating and asciidoc tagging.
>
> Haha. I know how this feels. It is often tough not to change something. I
> was even intentionally inconsistent about the "*" being directly at the
> beginning of a line or indented by one to minimize the changes that I would
> need to do. Just the indenting I fixed.
>
Okkk, I was wondering why 😅 But there again I couldn't help but making
suggestions in my review!
Another point that triggers OCD is anchoring and index tagging and you'll
find a lot of suggestions on this point too.

Like I told you in matrix/irc PM, could you (and others) please make me a
> halfway feedback to tell me what you think...
>
> I likely missed this if this is anything recent. Am mostly offline these
> days.
>
Could you please just have a quick look at my in process review and tell me
if it's worth keeping on like this, or if I'd better just look for errors
only?


My next task will definitely be working on the docs writing guidelines to
> submit them, discuss them, and have them approved and merged ASAP. I won't
> invest any more time on this documentation code base before making it a
> solid foundation for the future LinuxCNC documentation system (yes, I'm
> looking at you Antora!) Always glad to be part of this and to collaborate
> with you guys
>
> Antora seems very interesting, indeed. Just if this PR does not go forward
> then everything else that is even more complicated I do not see happening
> either.
>
Having this PR blocked would be a critical/emergency warning me... And this
should be no surprise to anyone interested in this as I always said this,
and much more, should have been done first hand!

So, after making things in the wrong ordrer, and choosing a documentation
translation platform that's a pain for the translator without more common
decision than this, if we ultimately can't do what must be done, well...


I think we need to somehow meet and talk about po4a, an idea about how the
> documentation should look, and how to get there. A mini-conference would be
> nice.
>
I'm 1000% with you on this! And documentation is not the only mater where a
mini-conference followed by a statement of the status and future plan would
be immensely beneficial to the project and community building.

As I've learned the hard way this kind of message can have unwanted
effects, let me ensure the sincere respectfulness and good willingness of
it 🙏

TY
Jérémie

smoe commented:
>>> Le sam. 12 mars 2022 à 00:58, Steffen Möller ***@***.***> a écrit :
>>> While reviewing your PR, I see me again unable to refrain making a significant number of additional mods suggestions to try to get the src english doc to a clean sound formating and asciidoc tagging.

>>Haha. I know how this feels. It is often tough not to change something. I was even intentionally inconsistent about the "*" being directly at the beginning of a line or indented by one to minimize the changes that I would need to do. Just the indenting I fixed.

> Okkk, I was wondering why  But there again I couldn't help but making suggestions in my review!

I have looked around but could not find your comments.  Maybe we need to screenshare over this. I had done so with Petter and that was very fruitful to get me started back in the days ... feels like a long time ago :)

> Another point that triggers OCD is anchoring and index tagging and you'll find a lot of suggestions on this point too. 

_Yes_! Here I would truly appreciate some "that's how to do it perfectly right!"-kind of instruction. What I have come up with is from what I had read from the manuals and that has worked and everything else can just wait for the style guide.

[...]
> Could you please just have a quick look at my in process review and tell me if it's worth keeping on like this, or if I'd better just look for errors only?

I still need to find your comments. Just anything that brings this PR forward towards its acceptance would be welcome. If you beautify more then the diff will increase, which may render it more difficult to get accepted. What we have here is the first po4a-compatible version that builds - after changes done to the _es files like removing stale references.

The motivation to go through all the other files without translations was only to allow po4a to do some decent parsing of the text, with the experiences made from the files that had that _es counterpart, since otherwise, well, we'd only get the first line of what is a multi-line item in an itemization - have fun translating that. It is the first time I am explicitly writing this somewhere since this was so very obvious to me who I synced the _es files. Maybe I should put in bold that **this PR is a precondition to even think about using po4a and not offending the weblate community** with requests to translate nonsensical fragments of sentences/paragraphs?

Besides its size, I do not know why this PR is not immediately accepted. It is manually linting the .adocs. Yeah. Great. My hunch is that folks would like to hear that this PR indeed does no harm and just be accepted. From this we would then regenerate the .po files for Petter and while looking at the fragmentation level of those .po files which should then build. Afterwards we will skim through the .po files again and when these look badly fragmented decide to add extra fixes in another iteration.

>>> My next task will definitely be working on the docs writing guidelines to submit them, discuss them, and have them approved and merged ASAP.

Maybe give a pointer to your efforts in this thread. I had once started something in a README somewhere - but whatever you have - yes, please!

>>> I won't invest any more time on this documentation code base before making it a solid foundation for the future LinuxCNC documentation system (yes, I'm looking at you Antora!) Always glad to be part of this and to collaborate with you guys

>> Antora seems very interesting, indeed. Just if this PR does not go forward then everything else that is even more complicated I do not see happening either.

> Having this PR blocked would be a critical/emergency warning me... And this should be no surprise to anyone interested in this as I always said this, and much more, should have been done first hand! So, after making things in the wrong order, and choosing a documentation translation platform that's a pain for the translator without more common decision than this, if we ultimately can't do what must be done, well... 

I am at ease with all decisions taken - and I do not feel that I was not involved in them - at least I decided to help which I did not need to do. Also, if we can move from one translation-community to another without too much of a hassle then I do not see too much of a risk and these .po files are a standard that I do not mind to support and prepare for. I am just unhappy with stalling things now.

>> I think we need to somehow meet and talk about po4a, an idea about how the documentation should look, and how to get there. A mini-conference would be nice.

> I'm 1000% with you on this!

Let's have one, then. I'll ask "my three main LinuxCNC anchors", you and Petter for a joint slot for a documentation mini-conf and they/you know whom else to ask to join.

> And documentation is not the only mater where a mini-conference followed by a statement of the status and future plan would be immensely beneficial to the project and community building. As I've learned the hard way this kind of message can have unwanted effects, let me ensure the sincere respectfulness and good willingness of it.

It is a very natural thing that those who are leading the project do not necessarily know that they are leading. And leadership is something you need if you want your spare time to be used efficiently - and be fruitful.

andypugh commented:
I hadn't realised that the authors of this PR had considered it "blocked". 
If it hasn't been merged then I don't think that this is a sign that it is not approved of, just that everyone on the approvers list has assumed that someone else understands it better than they do. 

At this point I would like to see the conflict that is being flagged above resolved, but once it passes the tests I am happy to take the author's word for it that it is ready to merge.

smoe commented:
Happy! I'll address the conflict, sync with @silopolis and then send a ping.

smoe commented:
@silopolis , you mentioned comments/changes that I cannot find. For those and (sudden idea) the changes you have prepared to complete the po4a-ication of the non-trivial _fr files, can you merge (or preferably rebase on) my branch with the changes you have prepared and send me a PR against my branch?

hansu commented:
The merge conflict is caused by a commit of mine, sorry. But should be easy to resolve. I tried to resolve this with the GitHub UI, but I could only apply the change when also merging this.

silopolis commented:
Hey,


So glad to see things moving!

Le lun. 14 mars 2022 à 01:43, Steffen Möller ***@***.***> a
écrit :

> @silopolis <https://github.com/silopolis> , you mentioned
> comments/changes that I cannot find.
>
Seems because I started a review on GH instead off making direct comments :/

> For those and (sudden idea) the changes you have prepared to complete the
> po4a-ication of the non-trivial _fr files, can you merge (or preferably
> rebase on) my branch with the changes you have prepared and send me a PR
> against my branch?
>
So IIUC, you want me to rebase my work on the former @pere 's
translate-po4a branch onto yours, with both french and english
modifications, right?

smoe commented:
@silopolis , you understood me right but now I am not 100%ish sure what is best. At the moment, nothing of @pere's po4a infrastructure is part of that changes* branch. So I would not want those scripts of his in - the .po generation from the _es and _fr files we will never do again. But the changes you have done to the _fr files I would like to have.
The way I had done it was to sync with master and then clean the diff from the po4a infrastructure and create a new, i.e. this "changes.." branch, from it and then try to get it to build again (took a while, darn links).

So, while I was typing, becoming increasingly aware that my _es- and your _fr-motivated changes to the _en files also have diverged, maybe it is best if we get this set of changes of mine into master and the we repeat that process for your _fr-motivated set. Then maybe another iteration to iron out last problems with the _po generation once we have everything together. The style guide and weblate we can then address in parallel.

Now, I also checked https://github.com/smoe/linuxcnc/tree/changes_on_English_for_weblate but could not find any comment/fix of yours. Would you have a URL for me? Or can you generate a diff from what you have access to?

smoe commented on docs/README.adoc:
```diff
@@ -122,6 +122,87 @@ Drawings from QCAD
  - Export the  image as 480 x 480 Resolution Auto .png
 
 
+Dealing with translations
```
I suggest to postpone this translations.adoc until we know what we are doing, but yet, this would be good.

smoe replied:
Can you please save this TODO somewhere?

silopolis replied:
There https://tree.taiga.io/project/silopolis-silopolislinuxcnc/issue/27

smoe commented on docs/src/Master_Developer.adoc:
```diff
@@ -1,37 +1,25 @@
-====================================
-
-:lang: en
+:ascii-ids:
```
No idea. Is this something I did?

silopolis replied:
Looks like so lol

smoe replied:
I found https://groups.google.com/g/asciidoc/c/MnO-Cs1Xhgo which speaks against using this. Apparently I had liked this once of have seen it in the translation without thinking too much about it.

smoe replied:
Another TODO, I suggest.

silopolis replied:
Here https://tree.taiga.io/project/silopolis-silopolislinuxcnc/issue/28

silopolis commented on docs/README.adoc:
```diff
@@ -122,6 +122,87 @@ Drawings from QCAD
+Dealing with translations
+-------------------------
+
+Translation of our documentation is in transition at the moment.
```
Added this because I can't tell how many time in 20 years I've seen "at the moment" in FOSS softwares docs where it was at least unclear when that moment actually was, and at worst clearly a looong time ago.

smoe replied:
Then we have "git blame" - nah, admittedly awkward but just take it as a placeholder that something needs to be done.

smoe commented on docs/src/getting-started/getting-linuxcnc.adoc:
```diff
@@ -235,15 +233,17 @@ to LinuxCNC when you go on line and allow you to easily upgrade with no
+during the Live CD install it cannot recognize the hard drive
 during the boot up.
 
+[[sec:_alternate_install_methods]]
```
Are you sure? "sec" is used frequently as an abbreviation of "section" throughout the documentation. I am unlikely to have added this just for the fun of it, well, not completely impossible, but, ...

silopolis replied:
It is indeed, but I can't see the why !? On the contrary, putting a prefix makes more sense to me, like in the G and M codes documents.
For doc writer, this also eases using internal links if they follow a meaningful pattern

smoe replied:
This is something for "after the style-guide" which is something for "after po4a".

smoe commented on docs/src/common/linux-faq_es.adoc:
```diff
@@ -1,13 +1,14 @@
 las páginas del manual con el comando man.
 
-== Login automatico (((Automatic Login)))
+== Login automatico(((Automatic Login)))
```
Good point. Please save this for the style guide.

smoe commented on docs/src/common/linux-faq_es.adoc:
```diff
@@ -74,9 +73,7 @@ directorio.
-== Paginas de manual [[sec:Man-Pages]]
-
-(((Man Pages)))
+== Paginas de manual[[sec:Man-Pages]](((Man Pages)))
```
Whatever works reliably is fine with me.

smoe commented on docs/src/config/core-components.adoc:
```diff
@@ -320,61 +344,31 @@ and `spindle-orient-mode`.
+* 'motion.debug-float-3' - (float, RO) This is used for debugging purposes.
+* 'motion.debug-s32-0' - (s32, RO) This is used for debugging purposes.
+* 'motion.debug-s32-1' - (s32, RO) This is used for debugging purposes.
+* 'motion.servo.last-period' - (u32, RO) The number of CPU cycles between invocations of the servo
+  thread. Typically, this number divided by the CPU speed gives the time
+  in seconds, and can be used to determine whether the realtime motion
+  controller is meeting its timing constraints
```
Ouch, no!  Why?

silopolis replied:
May be ignored, but I think I've read that this is against good formating. Have to check my style guide draft again...

smoe commented on docs/src/config/ini-config.adoc:
```diff
@@ -234,215 +228,175 @@ Descriptions of the interfaces are in the Interfaces section of the
-  select. 0.5 means 50% of the programmed spindle speed. (This is used to
-  set the minimum spindle speed). On multi spindle machine there will be entries for each spindle number. Qtvcp only
-
+* 'MIN_SPINDLE_0_OVERRIDE = 0.5' - The minimum spindle override the user may
+  select. 0.5 means 50% of the programmed spindle speed.  (This is used to
+  set the minimum spindle speed). On multi spindle machine there will be entries for each spindle number. Qtvcp only.
```
Again, this ruins the item!

silopolis commented on docs/src/common/emc-history_es.adoc:
```diff
@@ -1,7 +1,6 @@
 ﻿:lang: es
 
 [[cha:linuxcnc-history]]
```
Can't remember. 
Can be added in weblate later

smoe commented:
@silopolis - I am not ultimately happy. For one - this is far too fine-grained. Every commit triggers are rebuild. And there are 275 of them.  When you add references - these are your french set of changes? If not then I do not want to add anything that is not used that I would also need to add to _es and _fr to get po4a to work again.  And we do not 275 trivial commits to bisect through in case of an error surfacing somewhere, so I would rebase and squash them, still - this is all not working.

The itemizations I have fixed with the minimal number of changes required. You wished you would not have touched them, at least not without a clear policy that we indeed want the "*" to be at the very beginning of the line. I was always happy if I could fix the indentation by introducing a blank in the beginning. 

So, what drives me is that we need to have the po4a working before we can apply whatever style guide you have in mind. Any change that does not help po4a I would like to postpone.

smoe commented on docs/src/drivers/pluto-p.adoc:
```diff
@@ -17,80 +18,80 @@ The Pluto P board requires EPP mode. Netmos98xx chips do not work in EPP mode.
 === Connectors
 
-* The Pluto-P board is shipped with the left connector presoldered, with
+ * The Pluto-P board is shipped with the left connector presoldered, with
```
Further changes to indenting after the style guide emerged.

smoe commented on docs/src/drivers/pluto_p_fr.adoc:
```diff
@@ -1,3 +1,5 @@
+:lang: en
+
 = Pluto-P
```
Yes, the file should go.

smoe commented on docs/src/examples/gs2-example.adoc:
```diff
@@ -13,8 +17,6 @@ screen.
 In the custom.hal file we place the following to connect LinuxCNC to the
 GS2 and have LinuxCNC control the drive.
-
-.GS2 Example
```
I had problems with po4a - am not exactly sure why this now was removed.

smoe replied:
po4a could not distinguish the block titles with image titles or so.

smoe commented on docs/src/gcode/coordinates.adoc:
```diff
@@ -23,11 +28,11 @@ machine coordinate system.
 
 == Coordinate Systems
 
+[[fig:decalages-ilots]]
```
This looks much like coming from the Spanish document.

smoe commented:
This fails now because of https://github.com/smoe/linuxcnc/runs/5540020670?check_suite_focus=true#step:5:6207 - some "----" delimiter not matched (likely the one I was wondering about while accepting it). I'll fix that in an instant.

At least we now know that this kind of changes/comments made to the documentation is not working. Never again if not trivial, please. The changes by @silopolis are all meaning well, but they also introduce new references/indexes which will cause trouble with po4a, which then delays my plan to help Petter as soon as possibe. I am half-through with the changes, which is a hassle to get to because I need to first show every other 15 hidden ones.

smoe commented:
;) So, while working with these changes, the top separator made it into the first and the bottom separator into the second, so the build error and the email I received for it only referred to the first commit. :o/

smoe commented on docs/src/gcode/machining-center.adoc:
```diff
@@ -145,61 +155,60 @@ feed rate, or at traverse rate, or it may be synchronized to the
-[[sub:feed-rate]](((Feed Rate)))
-
-=== Feed Rate
+[[sub:feed-rate]]
```
Style-guide!

smoe commented on docs/src/gcode/machining-center.adoc:
```diff
@@ -145,61 +155,60 @@ feed rate, or at traverse rate, or it may be synchronized to the
    units in the ABC 'pseudo-cartesian' system.
 
-=== Coolant (((coolant)))
+=== Coolant(((coolant)))
```
Joined the two. Thanks!

smoe commented on docs/src/gcode/o-code.adoc:
```diff
@@ -26,8 +32,7 @@ o100 endsub
 ----
 
 [[ocode:comments]]
```
No, this is referenced in the singlar form and I acually think that this is preferable.

smoe commented on docs/src/gcode/o-code.adoc:
```diff
@@ -39,13 +44,12 @@ The behavior is undefined if:
 [[ocode:subroutines]]
-== Subroutines
-(((Subroutines)))
+== Subroutines(((Subroutines)))
```
-> singular

smoe commented on docs/src/gcode/o-code.adoc:
```diff
@@ -80,21 +84,19 @@ o100 sub
 o100 endsub
 ----
-See the <<gcode:binary-operators,Binary Operators>> & <<gcode:parameters,Parameters>> sections for more information.
+See the <<gcode:binary-operators,Binary Operators>> and <<gcode:parameters,Parameters>> sections for more information.
```
-> singular

smoe commented on docs/src/gcode/machining-center.adoc:
```diff
@@ -244,18 +271,15 @@ Interpreter know what the setting of any of these switches is.
-information.
+See the <<mcode:m48-m49,M48 M49 Override>> section for more information.
 
 [[sub:block-delete-switch]]
```
style guide

smoe commented on docs/src/gcode/machining-center.adoc:
```diff
@@ -244,18 +271,15 @@ Interpreter know what the setting of any of these switches is.
 
 [[sub:block-delete-switch]]
-
 === Block Delete Switch
```
Style-guide  / some general consensus about how much we want to have in the index.

smoe commented on docs/src/gcode/machining-center.adoc:
```diff
@@ -14,34 +16,40 @@ Interpreter. Mechanical components that do not interact directly with
-This is called 'kinematics'.]
+This is called 'kinematics'.
+
+NOTE:
```
style guide

smoe commented on docs/src/gcode/other-code.adoc:
```diff
@@ -15,8 +16,7 @@ or 'feed per revolution mode' are in effect, in which case the feed rate
 is as described in the <<gcode:g93-g94-g95,G93 G94 G95>> Section.
 
 [[sec:set-spindle-speed]]
```
style guide

smoe commented on docs/src/gcode/o-code.adoc:
```diff
@@ -26,8 +32,7 @@ o100 endsub
 [[ocode:comments]]
-== Comments
-(((Comments)))
+== Comments(((Comments)))
```
-> singular

smoe commented on docs/src/gcode/o-code.adoc:
```diff
@@ -39,13 +44,12 @@ The behavior is undefined if:
+that might have been mistyped. For example 'o100' is easier to
+see than 'O100' that it is not a '0'.
 
 [[ocode:subroutines]]
```
-> singular

smoe commented on docs/src/gcode/overview.adoc:
```diff
@@ -237,7 +235,6 @@ are read/write (can be assigned a value).
 
 [[sub:numbered-parameters]]
-
 === Numbered Parameters
```
style guide

smoe commented on docs/src/getting-started/getting-linuxcnc.adoc:
```diff
@@ -129,7 +127,7 @@ https://www.raspberrypi.org/documentation/installation/installing-images/README.
+used to boot a computer. The image is too large to fit on a CD.
 
 .Writing the image to a USB storage device in Linux
 
```
style guide - i prefer the separation

silopolis commented:
Le lun. 14 mars 2022 à 15:35, Steffen Möller ***@***.***> a
écrit :

> @silopolis <https://github.com/silopolis> - I am not ultimately happy.
>
Neither do I as it turns out!

> For one - this is far too fine-grained. Every commit triggers are rebuild.
> And there are 275 of them.
>
If I could have imagined such behaviour, I wouldn't even have given that GH
reviewing system 😒
At least now we now it shouldn't be used!

When you add references - these are your french set of changes? If not then
> I do not want to add anything that is not used that I would also need to
> add to _es and _fr to get po4a to work again. And we do not 275 trivial
> commits to bisect through in case of an error surfacing somewhere, so I
> would rebase and squash them, still - this is all not working.
>
At this point I could drop all this review considering GH behaviour and
content drift WRT main objective...

So, what drives me is that we need to have the po4a working before we can
> apply whatever style guide you have in mind. Any change that does not help
> po4a I would like to postpone.
>
All in with that! Please drop everything not useful for po4a. I start over
once it will be running smooth and the style guide merged.

Sorry for the noise and burden 🙏

smoe commented:
I rebased to master and squashed commits to the same file into one. I consider this ready to review and potentially also to upload.  There two next things:
 a) an equal set of changes by @silopolis that is backpropagating where the _fr documentation has outperformed the _en
 b) me regenerating the .po files and likely doing a few changes to the _es/_fr files in that process, like having placeholders for the parts that _fr has already contributed.

silopolis commented on docs/src/drivers/gs2.adoc:
```diff
@@ -1,53 +1,54 @@
 
 This is a userspace HAL program for the GS2 series of VFD's at
-Automation Direct.
+Automation Direct. footnote:[En Europe on trouve l'équivalent sous la marque Omron.]
```
```suggestion
Automation Direct. footnote:[In Europe equivalent hardware  can be found under the Omron brand.]
```

silopolis commented on docs/src/drivers/gs2.adoc:
```diff
@@ -1,53 +1,54 @@
 The above command says: loadusr, wait for named to load, 
-component gs2_vfd, named spindle-vfd
+component gs2_vfd, named spindle-vfd.
+FIXME from French: La commande de HAL _loadusr_ est détaillée au 
+chapitre: 
```
```suggestion
```
Translated the line just after

silopolis commented:
Hi folks,

After discussion with Steffen, and as next steps will trigger another
couple of full passes on docs, please consider this ready to merge.

This will allow to unlock work on translations migration to po4a to have it
finally pushed and merged.

Thanks in advance

Wishing yall a beautiful we
TY
Jérémie

smoe commented:
@silopolis , for the style guide I found that po4a already makes some suggestions about indenting what continues the previous line (https://github.com/LinuxCNC/linuxcnc/runs/5265160863?check_suite_focus=true#step:5:21152). The style guide could have some section like "po4a should not complain" and then below explain what this means.

silopolis commented:
Le mar. 22 mars 2022 à 01:25, andypugh ***@***.***> a écrit :

> Merged #1516 <https://github.com/LinuxCNC/linuxcnc/pull/1516> into master.
>

🎉🎉🎉
Thanks Andy 👍🙏

andypugh commented:
I was expecting this to fix the merge conflicts in https://github.com/LinuxCNC/linuxcnc/pull/1670 
Perhaps we need to pike GitHub in some way to re-run the checks?

smoe commented:
> I was expecting this to fix the merge conflicts in #1670 Perhaps we need to poke GitHub in some way to re-run the checks?

I just rebased the "EnglishAnchors"-branch (#1670) and got a series of warnings about git cherry-picks and prior-applied commits that would be skipped, but no conflicts. No idea what is happening - an artifact of me having squashed a few smaller patches?

I have just done another interactive merge of #1670 against master and squashed a few smaller patches into one, also corrected a few typos in the commit descriptions.

hansu commented on docs/src/config/core-components.adoc:
```diff
@@ -63,212 +64,237 @@ of the low level motion planner.
+  - 5: Probing
+  - 6: Rotary axis indexing
+
+* 'motion.on-soft-limit' - (bit, out) TRUE when the machine is on a soft limit.
+
+* 'motion.probe-input' - (bit, in)
+  'G38.n' uses the value on this pin to determine when the probe has made contact.
+  TRUE for probe contact closed (touching), FALSE for probe contact open.
+
+* 'motion.program-line' - (s32, out) The current program line while executing.
+  Zero if not running or between lines while single stepping.
+
+* 'motion.requested-vel' - (float, out)
+  The current requested velocity in user units per second.
+  This value is the F-word setting from the G-code file, possibly reduced to accommodate machine velocity and acceleration limits.
+  The value on this pin does not reflect the feed override or any other adjustments.
+
+* 'motion.teleop-mode' - (bit, out) TRUE when motion is in 'teleop mode', as opposed to 'coordinated mode'
+
+* 'motion.tooloffset.x ... motion.tooloffset.w' - (float, out, one per axis) shows the tool offset in effect;
+  it could come from the tool table ('G43' active), or it could come from the gcode ('G43.1' active)
+
+* 'spindle.0.at-speed' - (bit, in)
+  Motion will pause until this pin is TRUE, under the following conditions:
+** before the first feed move after each spindle start or speed change;
+** before the start of every chain of spindle-synchronized moves;
+** and if in CSS mode,
+  at every rapid to feed transition.
+  +
+  This input can be used to ensure that the spindle is up to speed before starting a cut, or that a lathe spindle in CSS mode has
+  slowed down after a large to small facing pass before starting the next pass at the large diameter.
+  Many VFDs have an 'at speed' output.
+  Otherwise, it is easy to generate this signal with the 'HAL near' component, by comparing requested and actual spindle speeds.
```
Where do you have this part from?

hansu replied:
@smoe

smoe replied:
Cannot recall. Was is possibly just moved somehow within the same file?

smoe commented:
I just googled for the fragment "TRUE for probe contact closed (touching), FALSE for probe contact open." in "http://linuxcnc.org/docs/html/man/man9/motion.9.html". Does this help?

smoe commented:
Other parts of that diff are also from motion.9.  Hm. Did I mess something up?

hansu commented:
Ah yes the man page, thanks :+1: 

>Did I mess something up?

Ähm not in this file. Just want to validate this. But I found some places where you re-introduced a bit obsolete documentation. It was removed in the english docs but not updated in the french or spanish. And your back-translation brought it in again :)

So I have to review the changes if this PR more carefully to not have obsolete docs at the end of the day. 
My plan was to translate the placeholder paragraphs. But somewhere in the middle I realized that I also had to check the content because something obviously obsolete caught my eye. So once again from the beginning :smile: 

Now I am at folder 'dir'. Seems not that I can finish this today.
And then once again have a look at Jérémies changes...
...would have been better to have had it in a separate branch.

smoe commented:
Aaah! I just checked the Spanish document and apparently I did a quick'n'dirty backtranslation (likely google-supported or so) of the corresponding paragraph. At the time I did not like that this was not counter-checked, so I then changed to leaving the original language whenever I encountered additional paragraphs.

smoe commented:
Darn. Yes, removal of content can be an improvement, too. I admittedly had not thought about that.

smoe commented:
Anyway - this very much explains why we are doing this po4a in the first place.

hansu commented:
>Darn. Yes, removal of content can be an improvement, too. I admittedly had not thought about that.

Yes, me too :smile: 

But while I am going through I imagine what hell of a work you have done to align the docs of the different languages.

smoe commented:
We need to have the _en in a decent-enough format for po4a to parse it and present reasonable sentences to the weblate community. Admittedly, I think we are at that stage now, mostly.

I see the generation of .po files for _fr and _es is a matter of respect to our volunteers who crafted those files. Their work should not be lost and will guide the community, I am sure. And we have something to show to our weblate community at day 0.

hansu commented:
Also we reintroduced parts which Sebastian removed five years ago in https://github.com/LinuxCNC/linuxcnc/commit/2f0d47b4dccb4aa610fcf0622fed8cbbcd2ca147

But this bug is probably not existing any more :D

silopolis commented:
Le mer. 6 avr. 2022 à 21:16, Hans ***@***.***> a écrit :

> Also we reintroduced parts which Sebastian removed five years ago in
> 2f0d47b
> <https://github.com/LinuxCNC/linuxcnc/commit/2f0d47b4dccb4aa610fcf0622fed8cbbcd2ca147>
>
> But this bug is probably not existing any more :D
>
I really hope so because it would mean we'd have to add a change of backend
to the roadmap!

Having an extensive and carefuly crafted index is of major importance.
